specs/codecs/dag-jose/fixtures/index.md

```diff
@@ -0,0 +1,157 @@
+DAG-JOSE Fixtures
+=================
+
+About this document
+-------------------
+
+This document contains test fixtures for the [DAG-JOSE](..) codec, and documentation describing them.
+
+These fixtures are executable by parsing them using the [testmark format](https://github.com/warpfork/go-testmark).
+If you're reading this file as markdown, or rendered to the web,
+know that all of the codeblocks here are data that can be easily programmatically extracted and used to drive tests in your implementation.
+
+These fixtures offer several different kinds of data:
+
+- DAG-JOSE serial data -- in hexidemical.  (DAG-JOSE is a binary format, so hexidemical is easier to work with here than the possibly-unprintable binary raw form.)
+	- These should be used as both an decode fixture, and as an encode fixture (the data should round-trip).
+- The list of paths that should be seen in that data, when parsed as [data model](/docs/data-model/).  (This is a good fixture because it's very human-readable, very easy to test, and also provides a form of documentation.)
+- Prettyprinted [DAG-JSON](/docs/codecs/known/dag-json/) data, which is the DAG-JOSE data, parsed, then re-encoded as DAG-JSON, and pretty-printed.  (This gives deep coverage, because it effectively examines every bit of the [data model](/docs/data-model/) view of the data.)
+
+Fixtures
+--------
+
+### JWS
+
+This is a DAG-JOSE object, in hexidemical:
+
+[testmark]:# (jws/serial.dag-jose.hex)
+```
+a2677061796c6f616458240171122089556551c3926679cc52c72e182a5619056a4727409ee93a26
+d05ad727ca11f46a7369676e61747572657381a26970726f7465637465644f7b22616c67223a2245
+64445341227d697369676e61747572655840fbff49e4e65c979955b9196023534913373416a11beb
+fdb256c9146903ddb9c450e287be379ca70a5e7bc039b848fb66d4bd5b96dae986941e04e7968d55
+b505
+```
+
+When it is parsed, we should see these paths within the data
+when we walk over it at the [data model](/docs/data-model/) level:
+
+[testmark]:# (jws/paths)
+```text
+link
```

Comment:
This line is one I particularly want to talk about.  Is this desired to be here -- aka, is "link" a field we want to have show up in the data model?  Or should it not be here, and is something that can sometimes be created by a helper function, that derives it on the fly?

(The question from https://github.com/ceramicnetwork/go-dag-jose/pull/23/files#r749773563, in other words.)

Reply:
Since `link` is deterministically derivable from `payload` and never actually encoded, I personally would have preferred it to come from a helper function that could just "enrich" decoded dag-jose data.

However, I couldn't leave it out of the schema and still `datamodel.Copy` an incoming Node (e.g. dag-json decoded) into the dag-jose builder to apply the latter's validations 😭 

(Just a little context to the decision in `go-dag-jose`)

Reply:
I could do something similar to the other logic I added that I don't quite like - that for "unflattening" JWE/JWS objects, for which I'm going from dag-json -> unflattened map -> json bytes -> reconstructed dag-json before passing that to dag-jose to encode. (BTW, is there an easier way to do that 😅?)

For decode, I could reconstruct the decoded data and add `link` after the decode but that feels super ugly too 🤷🏼‍♂️

Reply:
> Since `link` is deterministically derivable from `payload` and never actually encoded, I personally would have preferred it to come from a helper function that could just "enrich" decoded dag-jose data or "sanitize" data to be encoded by removing it before the encode.

Right, exactly -- that's what I was asking after.  That's probably what I would've expected, a priori.

That said, I don't have any reason to have a strong opinion one way or the other.  As long as it's agreed that the block form doesn't have it, and the data model form does, then... the specification is specified, and that's my main concern.